internal/service/advancedcluster/common_advanced_cluster_test.go

```diff
@@ -30,3 +47,240 @@ func TestRemoveLabel(t *testing.T) {
+			expectedResult: Result{
+				response: "",
+				state:    "PENDING",
+				error:    nil,
+			},
+		},
+		{
+			name:          "Error in the API call: Neither HTTP 503 or 404",
+			mockError:     genericError,
+			mockResponse:  &matlas.Response{Response: &http.Response{StatusCode: 400}, Links: nil, Raw: nil},
+			expectedError: true,
+			expectedResult: Result{
+				response: nil,
+				state:    "",
+				error:    genericError,
+			},
+		},
+		{
+			name:          "Successful but with at least one cluster not idle",
+			mockCluster:   &matlas.AdvancedClustersResponse{Results: advancedClusters},
+			mockResponse:  &matlas.Response{Response: &http.Response{StatusCode: 200}, Links: nil, Raw: nil},
+			expectedError: false,
+			expectedResult: Result{
+				response: advancedClusters[0],
+				state:    "PENDING",
+				error:    nil,
+			},
+		},
+		{
+			name:          "Successful",
+			mockCluster:   &matlas.AdvancedClustersResponse{},
+			mockResponse:  &matlas.Response{Response: &http.Response{StatusCode: 200}, Links: nil, Raw: nil},
+			expectedError: false,
+			expectedResult: Result{
+				response: &matlas.AdvancedClustersResponse{},
+				state:    "IDLE",
+				error:    nil,
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			testObject := new(MockClusterService)
+
+			response := ClusterResponse{
+				advancedClusterResponse: tc.mockCluster,
+				response:                tc.mockResponse,
+				error:                   tc.mockError,
+			}
+			testObject.On("List", mock.Anything, mock.Anything, mock.Anything).Return(response)
+
+			result, stateName, err := advancedcluster.ResourceClusterListAdvancedRefreshFunc(context.Background(), dummyProjectID, testObject)()
+			if (err != nil) != tc.expectedError {
+				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
+			}
+
+			assert.Equal(t, tc.expectedResult.error, err)
+			assert.Equal(t, tc.expectedResult.response, result)
+			assert.Equal(t, tc.expectedResult.state, stateName)
+		})
+	}
+}
+
+type MockClusterService struct {
+	mock.Mock
+}
+
+func (a *MockClusterService) Get(ctx context.Context, groupID, clusterName string) (*matlas.Cluster, *matlas.Response, error) {
+	args := a.Called(ctx, groupID)
+	var response = args.Get(0).(ClusterResponse)
+	return response.cluster, response.response, response.error
+}
+
+func (a *MockClusterService) List(ctx context.Context, groupID string, options *matlas.ListOptions) (*matlas.AdvancedClustersResponse, *matlas.Response, error) {
+	args := a.Called(ctx, groupID)
+	var response = args.Get(0).(ClusterResponse)
+	return response.advancedClusterResponse, response.response, response.error
+}
+
+func (a *MockClusterService) GetAdvancedCluster(ctx context.Context, groupID, clusterName string) (*matlas.AdvancedCluster, *matlas.Response, error) {
+	args := a.Called(ctx, groupID)
+	var response = args.Get(0).(ClusterResponse)
+	return response.advancedCluster, response.response, response.error
+}
```

Comment:
Should we generate the MockService with mockery?

Reply:
You are right! Did this tests before we decided to use mockery, and copy pasted them from previous PR. I already changed it.